config/config.go

```diff
@@ -37,28 +68,39 @@ func (config *Config) Validate(path string) ([]string, error) {
 		return nil, utils.NewConfigValidationFieldRequiredError(path, "data_dir")
 	}
 
-	if config.DataRateMsec < 0 {
-		return nil, errors.New("cannot specify data_rate_msec less than zero")
-	}
-
 	if config.MapRateSec != nil && *config.MapRateSec < 0 {
 		return nil, errors.New("cannot specify map_rate_sec less than zero")
 	}
 
-	deps := config.Sensors
+	deps := []string{cameraName}
 
 	return deps, nil
 }
 
 // GetOptionalParameters sets any unset optional config parameters to the values passed to this function,
 // and returns them.
-func GetOptionalParameters(config *Config,
-	defaultDataRateMsec, defaultMapRateSec int, logger golog.Logger,
-) (int, int) {
-	dataRateMsec := config.DataRateMsec
-	if config.DataRateMsec == 0 {
-		dataRateMsec = defaultDataRateMsec
-		logger.Debugf("no data_rate_msec given, setting to default value of %d", defaultDataRateMsec)
+func GetOptionalParameters(config *Config, defaultLidarDataRateMsec, defaultMapRateSec int, logger golog.Logger,
+) (int, int, error) {
+	lidarDataRateMsec := defaultLidarDataRateMsec
+
+	// feature flag for new config
+	if config.UseNewConfig {
+		strCameraDataFreqHz, ok := config.Camera["data_frequency_hz"]
+		if !ok {
+			logger.Debugf("problem retrieving lidar data frequency, setting to default value of %d", 1000/defaultLidarDataRateMsec)
```

Comment:
We don't know if this is a problem, lets rephrase:
```suggestion
			logger.Debugf("config did not provide camera[data_frequency_hz], setting to default value of %d", 1000/defaultLidarDataRateMsec)
```